invisible_cities/cities/detsim_test.py

```diff
@@ -0,0 +1,277 @@
+        assert hasattr(h5out.root, 'Filters/active_hits')
+
+        pmtrd  = h5out.root.pmtrd
+        sipmrd = h5out.root.sipmrd
+
+        n_pmt  = int(buffer_params["length"] / buffer_params["pmt_width"] )
+        n_sipm = int(buffer_params["length"] / buffer_params["sipm_width"])
+        assert pmtrd .shape == (1,   12, n_pmt)
+        assert sipmrd.shape == (1, 1792, n_sipm)
+
+
+def test_detsim_filter_active_hits(ICDATADIR, output_tmpdir):
+
+    PATH_IN  = os.path.join(ICDATADIR    , "nexus_new_kr83m_fast.newformat.sim.h5")
+    PATH_OUT = os.path.join(output_tmpdir, "detsim_test.h5")
+    conf = configure('detsim $ICTDIR/invisible_cities/config/detsim.conf'.split())
+
+    # modify labels for event_id = 0 in test file
+    # all its hits labels are set to "NO_ACTIVE"
+    modified_testfile = os.path.join(output_tmpdir, "modified_testfile.h5")
+    shutil.copyfile(PATH_IN, modified_testfile)
+    with tb.open_file(modified_testfile, "a") as h5file:
+        hits     = h5file.root.MC.hits
+        readhits = hits.read()
+        event0 = readhits[readhits["event_id"]==0]
+        hits.modify_column(start=0, stop=len(event0), column=["NO_ACTIVE"]*len(event0), colname="label")
+        hits.flush()
+
+    conf.update(dict(files_in    = modified_testfile,
+                     file_out    = PATH_OUT,
+                     run_number  = 0))
+    result = detsim(**conf)
+
+    assert result.events_in   == 2
+    assert result.evtnum_list == [1]
+
+    with tb.open_file(PATH_OUT, mode="r") as h5out:
+        filters = h5out.root.Filters.active_hits.read()
+        np.testing.assert_array_equal(filters["passed"], [False, True])
+
+
+def test_detsim_empty_input_file(ICDATADIR, output_tmpdir):
+
+    PATH_IN  = os.path.join(ICDATADIR    , "nexus_new_kr83m_fast.newformat.sim.h5")
+    PATH_OUT = os.path.join(output_tmpdir, "detsim_test.h5")
+    conf = configure('detsim $ICTDIR/invisible_cities/config/detsim.conf'.split())
+
+    # remove test file rows
+    empty_testfile = os.path.join(output_tmpdir, "empty_testfile.h5")
+    shutil.copyfile(PATH_IN, empty_testfile)
+    with tb.open_file(empty_testfile, "a") as h5file:
+        h5file.root.MC.configuration.remove_rows(0)
+        h5file.root.MC.hits         .remove_rows(0)
+        h5file.root.MC.particles    .remove_rows(0)
+        h5file.flush()
+
+    conf.update(dict(files_in    = empty_testfile,
+                     file_out    = PATH_OUT,
+                     run_number  = 0))
+    result = detsim(**conf)
+
+    assert result.events_in   == 0
+    assert result.evtnum_list == []
+
+
+def test_detsim_exact(ICDATADIR, output_tmpdir):
+
+    PATH_IN   = os.path.join(ICDATADIR    , "nexus_new_kr83m_fast.newformat.sim.h5")
+    PATH_OUT  = os.path.join(output_tmpdir, "detsim_test.h5")
+    PATH_TRUE = os.path.join(ICDATADIR    , "detsim_new_kr83m_fast.newformat.sim.h5")
+    conf = configure('detsim $ICTDIR/invisible_cities/config/detsim.conf'.split())
+
+    buffer_params = dict()
```

Comment:
No need to initialize empty dictionary, just create the dictionary with all the values you need.  You could **copy** the  `conf['buffer_params']`already existing dictionary and modify it since the only modification seems to be `pmt_width`